docs/design/pattern_matching.md

```diff
@@ -309,27 +330,16 @@ fn G() {
 }
 ```
 
-Pattern matching precedes the initialization of the storage for any `var`
-patterns. An introduced variable is only initialized if the complete pattern
-matches.
-
-```carbon
-class X {
-  destructor { Print("Destroyed!"); }
-}
-fn F(x: X) {
-  match ((x, 1 as i32)) {
-    case (var y: X, 0) => {}
-    case (var z: X, 1) => {}
-    // Prints "Destroyed!" only once, when `z` is destroyed.
-  }
-}
-```
-
```

Comment:
Would it be useful to retain this example and explain what happens? (I think what happens is that `"Destroyed!"` may be printed either once or twice, depending on the value of `x`.)

Reply:
> Would it be useful to retain this example and explain what happens?

Sure, but I think it should go with the discussion of `match` rather than here.

>(I think what happens is that `"Destroyed!"` may be printed either once or twice, depending on the value of `x`.)

I think it should always print twice, regardless of the value of `x`. The cases are distinguished by the second element of the pair, which is `1 as i32`, so the second case always matches.

(I think Carbon might reserve the right to print a third time, if it chooses to pass `x` by copy, but hopefully we won't in practice.)